tests/chainer_tests/functions_tests/activation_tests/test_softmax.py

```diff
@@ -26,105 +25,55 @@
-        for i in numpy.ndindex(y_roll.shape[:-1]):
-            y_roll[i] /= y_roll[i].sum()
-
-        testing.assert_allclose(
-            y_expect, y.data, **self.check_forward_options)
-
-    def test_forward_cpu(self):
-        self.check_forward(self.x)
-
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x))
-
-    @attr.gpu
-    def test_forward_gpu_non_contiguous(self):
-        self.check_forward(
-            cuda.cupy.asfortranarray(cuda.to_gpu(self.x)))
-
-    @attr.gpu
-    def test_forward_gpu_no_cudnn(self):
-        self.check_forward(cuda.to_gpu(self.x), 'never')
-
-    def check_backward(self, x_data, gy_data, use_cudnn='always'):
-        def f(x):
-            return functions.softmax(x, axis=self.axis)
-
-        with chainer.using_config('use_cudnn', use_cudnn):
-            gradient_check.check_backward(
-                f, x_data, gy_data, dtype=numpy.float64,
-                **self.check_backward_options)
-
-    def test_backward_cpu(self):
-        self.check_backward(self.x, self.gy)
-
-    @attr.gpu
-    def test_backward_gpu(self):
-        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
-
-    @attr.gpu
-    def test_backward_gpu_non_contiguous(self):
-        self.check_backward(
-            cuda.cupy.asfortranarray(cuda.to_gpu(self.x)),
-            cuda.cupy.asfortranarray(cuda.to_gpu(self.gy)))
-
-    @attr.gpu
-    def test_backward_gpu_no_cudnn(self):
-        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy), 'never')
-
-    def check_double_backward(self, x_data, gy_data, ggx_data,
-                              use_cudnn='always'):
-        def f(x):
-            return functions.softmax(x, axis=self.axis)
-
-        with chainer.using_config('use_cudnn', use_cudnn):
-            gradient_check.check_double_backward(
-                f, (x_data,), (gy_data,), (ggx_data), dtype=numpy.float64,
-                **self.check_double_backward_options)
-
-    def test_double_backward_cpu(self):
-        self.check_double_backward(self.x, self.gy, self.ggx)
-
-    @attr.gpu
-    def test_double_backward_gpu(self):
-        self.check_double_backward(
-            cuda.to_gpu(self.x), cuda.to_gpu(self.gy), cuda.to_gpu(self.ggx))
-
-    @attr.gpu
-    def test_double_backward_gpu_no_cudnn(self):
-        self.check_double_backward(
-            cuda.to_gpu(self.x), cuda.to_gpu(self.gy), cuda.to_gpu(self.ggx),
-            'never')
+            x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        return x,
+
+    def forward(self, inputs, device):
+        x, = inputs
+        return functions.softmax(x, axis=self.axis),
+
+    def forward_expected(self, inputs):
+        x, = inputs
+        expected = numpy.exp(x)
+        expected = numpy.rollaxis(expected, self.axis, expected.ndim)
+        for i in numpy.ndindex(expected.shape[:-1]):
+            expected[i] /= expected[i].sum()
+        return expected.astype(x.dtype),
```

Comment:
After changing the axis with `rollaxis`, it needs to be returned to the original shape after the `ndindex` is iterated.

```expected = numpy.rollaxis(expected, expected.ndim - 1, self.axis)                                                                                                                                                                                             ```

Reply:
Thanks for the review @Crissman . Made the fixes! Shapes are now matching.

Reply:
Thanks @Crissman . Made the required changes! The shapes are compatible now.